### DISCUSSION TITLE
Ignore updateInternalDependencies when importing latest "*"

### DISCUSSION THREAD

**Discussion by ghost** [changesets/changesets#1041] [Q&A] [answered]
I am importing all internal dependencies at the latest version with "*".

Is there a way to ensure that these version references are not modified after running the version script?

**Comment by Andarist** [comment]
`*` should not be touched by `changesets version` **unless** you are using the prerelease mode. Prerelease versions don't satisfy `*` and we can't leave them alone as you'd end up with packages that wouldn't be quite usable after installation.

**Reply by ghost** [reply]
So I'm avoiding using prelease mode in my workflow as it sounds like things could get messy.

Basically, I'm looking to release alpha versions for when PR branches are pushed. I'm doing this by using a snapshot release. I'm running `changeset && changeset version --snapshot {branchName}-{commit}`

What i'm finding is that it's changing my "`*`" references on packages which have the changed package as a dependent. It's using the new snapshot version instead of leaving the "`*`"

---

I've also noticed that when I'm looking to version the package in accordance to the semver convention (once the PR is approved and ready to be merged to master), it's not versioning the package correctly, and instead always versions back at `0.0.0`

Full example:

`package-a`
version: `1.0.0` (this version is published)

I then create a PR. Create a snapshot version and publish with `changeset && changeset version --snapshot {branchName}-{commit}`. `package-a` is now versioned:

version: `0.0.0-MyBranch-abc123` (this version is published)

The list of published versions for the `package-a` is now:

- `package-a-0.0.0-MyBranch-abc123`
- `package-a-1.0.0`

Now I go create a semver version ready for prod by running `changeset && changeset version`. Regardless of whether I select a `major`, `minor` or `patch`, the package will always be versioned at `0.0.0`. I can see in the `CHANGELOG.md`, all versions are accounted for, so I would have assumed it would be bumped to version `1.1.0` had I selected `minor`.

CHANGELOG.md:

```
# package-a

## 0.0.0

### Minor Changes

- Test

## package-a-0.0.0-MyBranch-abc123

### Patch Changes

- Test versioning

## 1.0.0

### Major Changes

- Initial commit
```

---

Is this not the recommended workflow for alpha releases on PR branches?

**Reply by Andarist** [reply]
> What i'm finding is that it's changing my "*" references on packages which have the changed package as a dependent. It's using the new snapshot version instead of leaving the "*"

Right, so ye - even though snapshots and prereleases are different features in Changesets, they are the same for npm. `*` is not satisfied by versions with prerelease segments (and you create such using snapshot versions). We absolutely have to replace that - as otherwise, your snapshot releases would be unusable. 

> I've also noticed that when I'm looking to version the package in accordance to the semver convention (once the PR is approved and ready to be merged to master), it's not versioning the package correctly, and instead always versions back at 0.0.0

You can change that behavior with `useCalculatedVersion` - but I also see that you already figured this out.

**Comment by ghost** [comment, marked as answer]
So I think configuring the `snapshot` to use `useCalculatedVersion` seems to have resolved all these issues...
